pkg/image/qemu.go

```diff
@@ -33,48 +33,95 @@ const (
 	maxCPUSecs         = 30      //value from OpenStack Nova
 )
 
-// ConvertQcow2ToRaw converts a local qcou2 image to raw format
+// ConvertQcow2ToRaw converts a local qcow2 image to raw format
 func ConvertQcow2ToRaw(src, dest string) error {
-	err := execWithLimits("qemu-img", "convert", "-f", "qcow2", "-O", "raw", src, dest)
+	err := validate(src, "qcow2")
+	if err != nil {
+		return errors.Wrap(err, "image validation failed")
+	}
+
+	_, err = execWithLimits("qemu-img", "convert", "-f", "qcow2", "-O", "raw", src, dest)
 	if err != nil {
 		return errors.Wrap(err, "could not convert local qcow2 image to raw")
 	}
+
 	return nil
 }
 
-// ConvertQcow2ToRawStream converts an http accessible qcow2 image to raf format without locally caching the qcow2 image
+// ConvertQcow2ToRawStream converts an http accessible qcow2 image to raw format without locally caching the qcow2 image
```

Comment:
what do you think of having a single `ConvertQcow2ToRaw` func with an "optional" json arg?